pkg/doc/verifiable/presentation_jwt_proof_test.go

```diff
@@ -129,6 +130,167 @@ func TestNewPresentationFromUnsecuredJWT(t *testing.T) {
+	r.NoError(e)
+	r.NotNil(vcJWS)
+
+	t.Run("Presentation with VC defined as JWS", func(t *testing.T) {
+		// Create and encode VP.
+		vp := &Presentation{
+			Context: []string{
+				"https://www.w3.org/2018/credentials/v1"},
+			ID:     "urn:uuid:3978344f-8596-4c3a-a978-8fcaba3903c",
+			Type:   []string{"VerifiablePresentation"},
+			Holder: "did:example:ebfeb1f712ebc6f1c276e12ec21",
+		}
+		err := vp.SetCredentials(vcJWS)
+		r.NoError(err)
+
+		holderPubKey, holderPrivKey, err := ed25519.GenerateKey(rand.Reader)
+		r.NoError(err)
+
+		vpJWS, err := vp.JWTClaims([]string{}, true).MarshalJWS(
+			EdDSA, holderPrivKey, "holder-key")
+		r.NoError(err)
+
+		// Decode VP
+		vpDecoded, err := NewPresentation([]byte(vpJWS), WithPresPublicKeyFetcher(
+			func(issuerID, keyID string) (interface{}, error) {
+				switch keyID {
+				case "holder-key":
+					return holderPubKey, nil
+				case "issuer-key":
+					return issuerPrivKey.Public(), nil
+				default:
+					return nil, errors.New("unexpected key")
+				}
+			}))
+		r.NoError(err)
+		vpCreds, err := vpDecoded.MarshalledCredentials()
+		r.NoError(err)
+		r.Len(vpCreds, 1)
+
+		vcDecoded, _, err := NewCredential(vpCreds[0])
+		r.NoError(err)
+
+		r.Equal(vc.stringJSON(t), vcDecoded.stringJSON(t))
+	})
+
+	t.Run("Presentation with VC defined as VC struct", func(t *testing.T) {
+		// Create and encode VP.
+		vp := &Presentation{
+			Context: []string{
+				"https://www.w3.org/2018/credentials/v1"},
+			ID:     "urn:uuid:3978344f-8596-4c3a-a978-8fcaba3903c",
+			Type:   []string{"VerifiablePresentation"},
+			Holder: "did:example:ebfeb1f712ebc6f1c276e12ec21",
+		}
+		err := vp.SetCredentials(vc)
+		r.NoError(err)
+
+		holderPubKey, holderPrivKey, err := ed25519.GenerateKey(rand.Reader)
+		r.NoError(err)
+
+		vpJWS, err := vp.JWTClaims([]string{}, true).MarshalJWS(
+			EdDSA, holderPrivKey, "holder-key")
+		r.NoError(err)
+
+		// Decode VP
+		vpDecoded, err := NewPresentation([]byte(vpJWS), WithPresPublicKeyFetcher(SingleKey(holderPubKey)))
+		r.NoError(err)
+		vpCreds, err := vpDecoded.MarshalledCredentials()
+		r.NoError(err)
+		r.Len(vpCreds, 1)
+
+		vcDecoded, _, err := NewCredential(vpCreds[0])
+		r.NoError(err)
+
+		r.Equal(vc.stringJSON(t), vcDecoded.stringJSON(t))
+	})
+
+	t.Run("Failed check of VC due to invalid JWS", func(t *testing.T) {
+		// Create and encode VP.
+		vp := &Presentation{
+			Context: []string{
+				"https://www.w3.org/2018/credentials/v1"},
+			ID:     "urn:uuid:3978344f-8596-4c3a-a978-8fcaba3903c",
+			Type:   []string{"VerifiablePresentation"},
+			Holder: "did:example:ebfeb1f712ebc6f1c276e12ec21",
+		}
+		err := vp.SetCredentials(vcJWS)
+		r.NoError(err)
+
+		holderPubKey, holderPrivKey, err := ed25519.GenerateKey(rand.Reader)
+		r.NoError(err)
+
+		vpJWS, err := vp.JWTClaims([]string{}, true).MarshalJWS(
+			EdDSA, holderPrivKey, "holder-key")
+		r.NoError(err)
+
+		// Decode VP
+		_, err = NewPresentation([]byte(vpJWS), WithPresPublicKeyFetcher(
```

Comment:
you should check that presentation is nil

Reply:
Well, in case of expected error I usually just check error presence and error message. I will change this for all error expecting checks then.

Reply:
That would be perfect. In case of an error we should also check return value to make sure that it is nil (or empty)

Reply:
Done! @sandrask